build/build-docker-deploy.sh

```diff
@@ -11,7 +11,7 @@ set -euo pipefail
 # itself inside of docker passing "docker" as the argument causing the
 # commands in the if-branch to be executed within the docker
 # container.
-if [ "${1:-}" = "docker" ]; then
+if [ "${1-}" = "docker" ]; then
```

Comment:
The advanced bash guide is nice, but it's unofficial. Is there anything official? Whenever we get into lesser-known corners of bash I'm especially concerned about version differences because OSX is forever sticking with an ancient version of bash (Apple refuses to ship anything that uses GPLv3, which includes bash 4.x).

Oh, I see. Buried in the prose of the official docs is a note that "Omitting the colon results in a test only for a parameter that is unset." Subtle. The docs also confusingly use "null" as synonymous with "empty string".

http://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html